### DISCUSSION TITLE
Weird behaviour with TagsColumn and pivot tables

### DISCUSSION THREAD

**Discussion by Rossifer** [filamentphp/filament#3662] [General]
I have a table with a tags column that pulls from a roles pivot table. Generally this works well but occasionally the column is blank. However if I go and edit the record I can see the role is correctly displayed in this view. I've also verified that the data in the pivot table is correct using HeidiSQL.

The query to pull this data looks correct when I view it in Laravel's debugbar:

`select `roles`.*, `human_role`.`human_id` as `pivot_human_id`, `human_role`.`role_id` as `pivot_role_id` from `roles` inner join `human_role` on `roles`.`id` = `human_role`.`role_id` where `human_role`.`human_id` in (13, 1464, 9223372036854775807, 1, 1)`

but it adds the **in()** function at the end of the query which seems to cause the data to not be displayed. If I query the DB directly and drop the **in()** part completely then all the records are showing correctly. This last part of the query seems to cause the issue.

Can anyone help me figure this out?

**Comment by danharrin** [comment]
Where are the `13, 1464, 9223372036854775807` IDs from in your database? Are they the IDs of the records in your table?

**Reply by Rossifer** [reply]
This is the part about the query that really is confusing me as the IDs I'm using in my DB are all UUIDs. So I have no idea where those number come from as they seem to bear no relevance to the IDs.

I have about 2000 records so as I page through I can see the query changing here is another one with different numbers:

`select `roles`.*, `human_role`.`human_id` as `pivot_human_id`, `human_role`.`role_id` as `pivot_role_id` from `roles` inner join `human_role` on `roles`.`id` = `human_role`.`role_id` where `human_role`.`human_id` in (5598, 741, 81, 9606, 0, 0, 0, 0, 117, 12, 13, 1464, 9223372036854775807, 1, 1, 1, 2225492, 222, 258, 2873440, 28, 29, 2, 2, 2, 2, 3683, 3994459, 3, 3, 3, 3, 3, 3, 0, 42, 44, 44, 46452, 4879609, 4, 4, 4, 0, 4, 51, 53141, 54, 557528, 568)`

This is on a 50 record per page setting.

**Reply by danharrin** [reply]
Is your model correctly set up with `$keyType = 'string'`

**Comment by Rossifer** [comment]
You're a genius. How did I miss this from my model. Thank you!

**Reply by danharrin** [reply]
Honestly it was a guess, dont give me too much credit 😂
